docs/tutorials/building-your-first-bot.mdx

```diff
@@ -0,0 +1,342 @@
+    {
+        // Stops all bots and cleans up the RG SDK
+        RGBotServerListener.GetInstance()?.StopGame();
+    }
+
+}
+```
+
+Once this component is implemented, create an empty object within the scene, rename it to `RGBotSpawner` (just so we can find it later),
+and attach this `CharacterBotSpawnManager` component to that object. Position it within the scene wherever you want to spawn your bot 
+(**make sure to set the Y coordinate to 0 so your bot does not float!**), and drag `RGBotSpawner`'s own Transform into the 
+**Bot Spawn Point** entry of the RGBotSpawner's `CharacterBotSpawnManager` component. Finally, find the 
+**Player** prefab provided by the sample project (located within `Assets/FirstBotDemo/Demo/Prefabs/Player.prefab`), and drag 
+this into the **Rg Bot Prefab** entry. Regression Games now knows how to load and unload bots from your scene! Make sure to save the 
+scene.
+
+![Screenshot of the RGOverlayCanvas.](./img/building-your-first-bot/spawner_6.png)
+
+## Add state information to relevant GameObjects
+
+We are onto the final parts of the integration! We need to indicate the following within our game:
+
+* What game state our bots can see
+* What actions our bots can perform
+
+First, let's add the state information. Regression Games marks states for bot consumption using the 
+[`[RGState]` attribute](/integrating-with-unity/defining-states#3---rgstate-fieldpropertymethod-attribute).
+This attribute can be applied to any public method that returns a value or any variable within a `MonoBehaviour`, and
+it will automatically be sent to the bot.
+
+First, open the `Assets/FirstBotDemo/Demo/Scripts/Player/PlayerAttack.cs` file, and add the following method to the class, which
+will allow our bot to know when it is currently in an attack animation (make sure to import any missing imports, with
+`using RegressionGames;`):
+
+```csharp
+[RGState]
+public bool IsAttacking()
+{
+    return !string.IsNullOrEmpty(_attackAnimation) && _animator.GetBool(_attackAnimation);
+}
+```
+
+Then, within the `Assets/FirstBotDemo/Demo/Scripts/EnemyController.cs` file, add the following attributes to the existing `GetCurrentHp()`
+and `GetTotalHp()` methods:
+
+```csharp
+/*
+* Gets the enemies current hp
+*/
+[RGState("CurrentHealth")]
+public int GetCurrentHp()
+{
+    return _currentHp;
+}
+
+/*
+* Gets the enemy's total HP from its assigned info
+*/
+[RGState("MaxHealth")]
+public int GetTotalHp()
+{
+    if (!enemyInfo)
+    {
+        Debug.LogWarning($"{gameObject.name} is missing entity info. Please check the inspector and assign missing fields");
+        return default;
+    }
+    return enemyInfo.hp;
+}
+```
+
+That's it! The bot will now be able to see the current and max
+health of the enemy, and whether or not the player is attacking. See the full [state documentation](/integrating-with-unity/defining-states)
+to see all the ways you can define state within your game, including grabbing all of the state from a MonoBehaviour.
```

Comment:
```suggestion
for more information on defining states within your game.
```

Concerned that the "including grabbing all of the state from a MonoBehaviour" won't make sense to someone at this point in the tutorial since they haven't gotten to the point of actually _using_ states for anything.